crates/wit-component/src/printing.rs

```diff
@@ -990,66 +1115,223 @@ fn is_keyword(name: &str) -> bool {
     )
 }
 
+/// A visitor that receives tokens emitted by `WitPrinter`.
+pub trait Output {
+    /// A newline is added.
+    fn newline(&mut self);
+    /// A keyword is added. Keywords are hardcoded strings from `[a-z]`, but can start with `@`
+    /// when printing a [Feature Gate](https://github.com/WebAssembly/component-model/blob/main/design/mvp/WIT.md#feature-gates)
+    fn keyword(&mut self, src: &str);
+    /// A type is added.
+    fn r#type(&mut self, src: &str, kind: TypeKind);
+    /// A parameter name of a function, record or a named return is added.
+    fn param(&mut self, src: &str);
+    /// A case belonging to a variant, enum or flags is added.
+    fn case(&mut self, src: &str);
+    /// Generic argument section starts. In WIT this represents the `<` character.
+    fn generic_args_start(&mut self);
+    /// Generic argument section ends. In WIT this represents the '>' character.
+    fn generic_args_end(&mut self);
+    /// Called when a single documentation line is added.
+    /// The `doc` parameter starts with `///` omitted, and can be an empty string.
+    fn doc(&mut self, doc: &str);
+    /// A semicolon is added.
+    fn semicolon(&mut self);
```

Comment:
Do you think it might make sense to forward these methods to `str` by default? That way the implementation for `OutputString` could be a bit simpler perhaps